tests/test_scripts.py

```diff
@@ -29,109 +29,119 @@
     ShowIntegrationsScript,
 )
 from testing import MockPlace
-from . import (
-    DatabaseTest
-)
-
 
-class TestLibraryScript(DatabaseTest):
 
-    def test_libraries(self):
+class TestLibraryScript:
+    def test_libraries(self, db_session, create_test_library):
 
-        library = self._library(name="The Library")
-        ignored = self._library(name="Ignored Library")
+        library = create_test_library(db_session, library_name="The Library")
+        ignored = create_test_library(db_session, library_name="Ignored Library")
 
         class Mock(LibraryScript):
-            # Mock of LibraryScript that returns a special value
-            # when all_libraries is called.
+            # Mock of LibraryScript that returns a special value when all_libraries is called.
 
             all_libraries_return_value = object()
+
             @property
             def all_libraries(self):
                 return self.all_libraries_return_value
-        script = Mock(self._db)
+
+        script = Mock(db_session)
 
         # Any library can be processed if it's identified by name.
-        for l in library, ignored:
-            assert script.libraries(l.name) == [l]
+        for lib in library, ignored:
+            assert script.libraries(lib.name) == [lib]
+
         with pytest.raises(ValueError) as exc:
             script.libraries("Nonexistent Library")
+
         assert "No library with name 'Nonexistent Library'" in str(exc.value)
 
-        # If no library is identified by name, the output of
-        # all_libraries is used as the list of libraries.
+        # If no library is identified by name, the output of all_libraries is used as the list of libraries.
         assert script.libraries() == script.all_libraries_return_value
 
-    def test_all_libraries(self):
+        db_session.delete(library)
+        db_session.delete(ignored)
+        db_session.commit()
```

Comment:
Presumably this has always been necessary in this code base and I never noticed before, but I think we need to introduce the code from core which runs each individual test in a database transaction that gets rolled back at the end. I don't like having to add this kind of cleanup code to the end of the tests.